components/molecules/SearchSelect.jsx

```diff
@@ -0,0 +1,185 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import SearchSelectOptions from './SearchSelectOptions';
+import debounce from '../utils/debounce';
+
+class SearchSelect extends Component {
+    static propTypes = {
+        debounceTime: PropTypes.number,
+        isFocused: PropTypes.bool,
+        isLoading: PropTypes.bool,
+        onChange: PropTypes.func.isRequired,
+        onSelect: PropTypes.func.isRequired,
+        options: PropTypes.array,
+        placeholder: PropTypes.string,
+        renderLoader: PropTypes.func,
+        size: PropTypes.oneOf(['xs', 'sm', 'md'])
+    };
+
+    static defaultProps = {
+        size: 'sm',
+        placeholder: 'Search',
+        options: [],
+        isFocused: false,
+        isLoading: false,
+        debounceTime: 300
+    };
+
+    state = {
+        isOpen: false,
+        searchTerm: ''
+    };
+
+    componentWillUnmount() {
+        this._removeListeners();
+    }
+
+    _mapContainerRef = ref => (this._container = ref);
+
+    _mapBtnRef = ref => (this._clearBtn = ref);
+
+    _addListeners() {
+        document.addEventListener('click', this._eventOutside);
+        document.addEventListener('focus', this._eventOutside, true);
+    }
+
+    _removeListeners() {
+        document.removeEventListener('click', this._eventOutside);
+        document.removeEventListener('focus', this._eventOutside, true);
+    }
+
+    _eventOutside = ({ target }) => {
+        if (!this._container.contains(target)) {
+            this.setState({ isOpen: false });
+            this._removeListeners();
+        }
+    };
+
+    _showOnFocus = ({ target }) => {
+        if (
+            this.props.options.length &&
+            this._container.contains(target) &&
+            target !== this._clearBtn
+        ) {
+            this.setState({ isOpen: true });
+            this._addListeners();
+        }
+    };
+
+    handleOptionClick = item => {
+        const { onSelect } = this.props;
+        onSelect(item);
+        this.setState({
+            isOpen: false,
+            searchTerm: item.title
+        });
+    };
+
+    _clearSearch = event => {
+        event.stopPropagation();
+        this.setState({ searchTerm: '', isOpen: false });
+        this._removeListeners();
+    };
+
```

Comment:
It can be good to have a default search filter, as we have for SearchMultiSelect, to filter the options in the current options list.